onnxruntime/core/framework/execution_provider.cc

```diff
@@ -66,14 +66,19 @@ void IExecutionProvider::InsertAllocator(AllocatorPtr allocator) {
   const OrtMemoryInfo& info = allocator->Info();
   auto ite = mem_info_set_.find(info);
   if (ite != mem_info_set_.end()) {
-    ORT_THROW("duplicated allocator");
+    LOGS_DEFAULT(WARNING) << "duplicated allocator: " << info.ToString();
+    return;
```

Comment:
I'm a little concerned about the behavior change. Maybe we should keep InsertAllocator as-is, while adding a new interface of TryInsertAllocator that allows dups?